docs/security-self-assessment.md

```diff
@@ -0,0 +1,268 @@
+
+  - The credentials provided in the API call match the ones configured in the server.
+
+  - The payload of the API call:
+    - Matches the Protobuf API definitions.
+    - Validates the parameters have the proper structure, e.g. users need to be written this way 'user:<userid>'
+  
+**Writing an Authorization Model**
+
+OpenFGA Authorization Models should be Directed Acyclic Graphs. OpenFGA validates that the model does not have cycles.
+
+When a model is written a new version is created. The application needs to be configured to use the new version when needed.
+
+**Calling the Authorization Query endpoints**
+
+When the [/check](https://openfga.dev/api/service#/Relationship%20Queries/Check) and [/list-objects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) endpoints are called, OpenFGA needs to traverse the graph defined by the model and instated by the tuples. The graph can be very wide and deep. To protect the service from a DoSS, OpenFGA needs to bound the number of simultaneous paths it explores.
+
+**Upgrading OpenFGA Database Schema**
+
+When new version of OpenFGA are installed, it might imply to run a database migration in the target system. To avoid downtime, migrations need to be carefully written and planned.
+
+### Goals
+
+* Provide developers a flexible way to implement authorization in their applications.
+
+* Provide a high performance and scalable authorization service that can be used in any application component.
+
+* Allow centralized authorization decisions, decoupling policy enforcement from decision making. Allows different teams to implement authorization using a common framework that can be used in any application component.
+
+* Not to be coupled to any authentication technology. It only requires a way to identify the subject of the authorization (user, application, device, etc).
+
+### Non-Goals
+
+* Does not provide tools for end-users to manage groups, roles, permissions.
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.
+
+This document serves to provide OpenFGA users with an initial understanding of OpenFGA's security, where to find existing security documentation, OpenFGA plans for security, and general overview of OpenFGA security practices, both for development of OpenFGA as well as security of OpenFGA.
+
+This document provides OpenFGA maintainers and stakeholders with additional context to help inform the roadmap creation process, so that security and feature improvements can be prioritized accordingly.
+
+## Security Functions and Features
+
+See [Actors](#actors) and [Actions](#actions) for more detailed description of the critical actors, actions, and potential threats.
+
+### Critical
```

Comment:
These should be 

"Critical. A listing critical security components of the project with a brief description of their importance. It is recommended these be used for threat modeling. These are considered critical design elements that make the product itself secure and are not configurable. Projects are encouraged to track these as primary impact items for changes to the project.
Security Relevant. A listing of security relevant components of the project with brief description. These are considered important to enhance the overall security of the project, such as deployment configurations, settings, etc. These should also be included in threat modeling."